tfx_bsl/beam/run_inference.py

```diff
@@ -697,81 +785,111 @@ def setup(self):
 
-  def _check_elements(
-      self, elements: List[Union[tf.train.Example,
-                                 tf.train.SequenceExample]]) -> None:
+  def _check_elements(self) -> None:
     pass
 
   def _post_process(
-      self, elements: Union[Sequence[tf.train.Example],
-                            Sequence[tf.train.SequenceExample]],
+      self, elements: Mapping[Any, Any],
       outputs: Mapping[Text, np.ndarray]
   ) -> Iterable[prediction_log_pb2.PredictLog]:
+    if not self._io_tensor_spec.input_tensor_types:
+      raise ValueError('No valid tensor types.')
+    input_tensor_names = self._io_tensor_spec.input_tensor_names
     input_tensor_alias = self._io_tensor_spec.input_tensor_alias
+    input_tensor_types = self._io_tensor_spec.input_tensor_types
     signature_name = self._signatures[0].name
-    batch_size = len(elements)
+
+    if len(input_tensor_alias) != len(input_tensor_names):
+      raise ValueError('Expected to have one name and one alias per tensor')
+
+    include_request = True
+    if len(input_tensor_names) == 1:
+      serialized_examples, = elements.values()
+      batch_size = len(serialized_examples)
+      process_elements = serialized_examples
+    else:
+      for tensor_name, tensor in elements.items():
+        if not isinstance(tensor, np.ndarray):
+          include_request = False
+          break
+
+      if include_request:
+        batch_size = len(elements[input_tensor_names[0]])
+      else:
+        batch_size = elements[input_tensor_names[0]].shape[0]
+
     for output_alias, output in outputs.items():
       if len(output.shape) < 1 or output.shape[0] != batch_size:
         raise ValueError(
             'Expected output tensor %s to have at least one '
             'dimension, with the first having a size equal to the input batch '
             'size %s. Instead found %s' %
             (output_alias, batch_size, output.shape))
-    predict_log_tmpl = prediction_log_pb2.PredictLog()
-    predict_log_tmpl.request.model_spec.signature_name = signature_name
-    predict_log_tmpl.response.model_spec.signature_name = signature_name
-    input_tensor_proto = predict_log_tmpl.request.inputs[input_tensor_alias]
-    input_tensor_proto.dtype = tf.string.as_datatype_enum
-    input_tensor_proto.tensor_shape.dim.add().size = 1
+  
+    if include_request:
+      predict_log_tmpl = prediction_log_pb2.PredictLog()
+      predict_log_tmpl.request.model_spec.signature_name = signature_name
+      predict_log_tmpl.response.model_spec.signature_name = signature_name
+      for alias, tensor_name in zip(input_tensor_alias, input_tensor_names):
+        input_tensor_proto = predict_log_tmpl.request.inputs[alias]
+        input_tensor_proto.dtype = tf.as_dtype(input_tensor_types[alias]).as_datatype_enum
+        if len(input_tensor_alias) == 1:
+          input_tensor_proto.tensor_shape.dim.add().size = 1
+        else:
+          input_tensor_proto.tensor_shape.dim.add().size = len(elements[tensor_name][0])
```

Comment:
Why the dim size is len(elements[tensor_name][0]) instead of:
for s in elements[tensor_name][0].shape:
  input_tensor_proto.tensor_shape.dim.add().size = s

Reply:
we have an nd.array, I dont think we will have shape parameter